Xamarin.Forms.Core/Visual.cs

```diff
@@ -4,9 +4,20 @@ namespace Xamarin.Forms
 {
 	public static class VisualMarker
 	{
+		static bool _isMaterialRegistered = false;
+
 		public static IVisual MatchParent { get; } = new VisualRendererMarker.MatchParent();
 		public static IVisual Default { get; } = new VisualRendererMarker.Default();
 		public static IVisual Material { get; } = new VisualRendererMarker.Material();
+
+		internal static void RegisterMaterial() => _isMaterialRegistered = true;
+		internal static void MaterialCheck()
+		{
+			if (_isMaterialRegistered)
+				return;
+
+			Internals.Log.Warning("Visual", "Material needs to be registered on iOS by calling FormsMaterial.Init() after Forms.Init()");
```

Comment:
What does this look like on Android or UWP? Maybe we should wrap this in a platform check...

Because we know that iOS needs a NuGet, we can have this message, but for the other platforms, we probably need another message - such as "Material is not supported on this platform." or something along those lines.

With regards to my comment above, I wouldn't actual write any `if (android) then don't log` code, but rather have the android platform actually call init. That way, we don't do anything special in here.

Reply:
I opted to still just check for the platform in this code because we want the message to be different on iOS vs other platforms. We could add some hooks so iOS can register its own message but that seems like overkill